### PR TITLE
Code coverage of OBO Authentication

### PR DESCRIPTION
**Pull request by RyanL1997**
### Description
Code coverage of OBO Authentication
* Category (Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation)
Enhancement

### Issues Resolved
* Resolve https://github.com/opensearch-project/security/issues/3101

### Check List
- [x] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peternied** [comment]
@RyanL1997 it looks like spotless found some issues in this PR, can you fix those and push?  It should also clean up the failing tests as I don't think they were related to your change.

File: src/test/java/org/opensearch/security/http/OnBehalfOfAuthenticatorTest.java [hunk 1]
```diff
@@ -12,11 +12,7 @@
 package org.opensearch.security.http;
 
 import java.nio.charset.StandardCharsets;
-import java.util.Base64;
-import java.util.Collections;
-import java.util.Date;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
```
Comment (DarshitChanpura) [inline src/test/java/org/opensearch/security/http/OnBehalfOfAuthenticatorTest.java:15]: nit: `*` import.

**Review by DarshitChanpura — COMMENTED** [review]
Thank you @RyanL1997. Could you add some final % number for code coverage?

**Review by scrawfor99 — DISMISSED** [review]

**Comment by DarshitChanpura** [comment]
@RyanL1997 Is this ready for final review?

**Comment by RyanL1997** [comment]
Final numbers:
- OBOAuthenticator [cov](https://app.codecov.io/gh/opensearch-project/security/pull/3428/blob/src/main/java/org/opensearch/security/http/OnBehalfOfAuthenticator.java): 95.83%
- JWTVendor [cov](https://app.codecov.io/gh/opensearch-project/security/pull/3428/blob/src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java): 96.97%

File: src/test/java/org/opensearch/security/http/OnBehalfOfAuthenticatorTest.java [hunk 2]
```diff
@@ -89,6 +133,44 @@ public void testBadKey() {
         Assert.assertTrue(exception.getMessage().contains("The specified key byte array is 80 bits"));
     }
 
+    @Test
+    public void testWeakKeyExceptionHandling() throws Exception {
+        Appender mockAppender = Mockito.mock(Appender.class);
+        ErrorHandler mockErrorHandler = Mockito.mock(ErrorHandler.class);
+        Mockito.when(mockAppender.getHandler()).thenReturn(mockErrorHandler);
```
Comment (DarshitChanpura) [inline src/test/java/org/opensearch/security/http/OnBehalfOfAuthenticatorTest.java:140]: nit: you can add static imports for all Mockito methods

**Review by DarshitChanpura — APPROVED** [review]
Great stuff @RyanL1997 boosting the code coverage!

**Review by scrawfor99 — APPROVED** [review]

**Pull request merged**